A2A project launches Java SDK blog post (#2344)
quarkusio/quarkusio.github.io: _posts/2025-06-27-a2a-project-lanunches-java-sdk.adoc

---
layout: post

> title: 'A2A Project launches Java SDK'

Reviewer: ```suggestion
title: 'Quarkus and WildFly teams from Red Hat collaborating with Google on launch of A2A Java SDK'
```

or something of that sort in the title that will be shared :)

date: 2025-06-27
tags: release

> synopsis: 'We’re thrilled to announce the launch of the A2A Java SDK, made possible by WildFly and Quarkus teams’ contribution.'

Reviewer: ```suggestion
synopsis: 'We’re thrilled to announce the launch of the Agent2Agent Java SDK, made possible by WildFly and Quarkus teams’ contribution.'
```

> author: maeste

Reviewer 2: The `_data/authors.yaml` file will need an update as well.

---

== *A2A Project Launches Java SDK*

The agent revolution just took a massive leap forward! Following the
recent landmark https://developers.googleblog.com/en/google-cloud-donates-a2a-to-linux-foundation/[announcement that Google has donated the Agent2Agent
(A2A) protocol to the Linux Foundation], we’re thrilled to announce the

> launch of the *https://github.com/a2aproject/a2a-java[A2A Java SDK]*, made possible by WildFly and Quarkus 

Reviewer: ```suggestion
launch of the *https://github.com/a2aproject/a2a-java[A2A Java SDK]*, built by WildFly and Quarkus teams in close collaboration, and now contributed to the official A2A project.
```

teams’ contribution.

=== *A New Era Under Linux Foundation Stewardship*

The A2A protocol’s transition to the Linux Foundation represents more
than just a change of governance: it’s a commitment to vendor-neutral,

> community-driven innovation that ensures this critical interoperability

Reviewer: ```suggestion
community-driven innovation. Similar how WildFly and Quarkus both recently joined CommonHaus fouundation. This ensures A2A as a critical interoperability
```

Agents.`"

=== *Why Java SDK Matters*

Here’s where things get exciting from a technical perspective: *true
polyglot agent ecosystems*.

The agent landscape has been fragmented, with Python dominating AI/ML
workflows, JavaScript powering web-based agents, and Java serving as the
backbone of enterprise backend systems. Each language ecosystem operated
in isolation, creating invisible walls that limited the collaborative
potential of multi-agent systems.

Our Java SDK shatters these barriers by implementing the A2A protocol
specification natively in Java, enabling:

* *Enterprise-grade agent integration* with existing Java
infrastructure +
* *Seamless interoperability* between Java agents and those written in
Python, JavaScript, or any A2A-compatible language +
* *Production-ready deployments* leveraging Java’s mature ecosystem and
well-tested enterprise capabilities

Want to see this interoperability in action? Explore our
https://github.com/a2aproject/a2a-samples/tree/main/samples/multi_language/python_and_java_multiagent[multi-language
sample implementation&#44;] which demonstrates how Python and Java
agents collaborate seamlessly.

> === *Technical Excellence: The Mutiny-Zero Advantage*

Reviewer: ```suggestion
=== Reactive Streams without framework lock-in
```

trying to avoid skimmers to tie it to mutiny

> The Java SDK leverages *mutiny-zero* as its reactive foundation, a

Reviewer: correct me if im wrong - is the sdk reactive all the way ? or is it just parts of it that are reactive?

It would be good to at least highlight reactive streams is not enforced on you as a user - but available if/when you need it?

=== Technical Excellence: The Mutiny-Zero Advantage

And if you need your agent to be reactive, don't worry about the dependencies you are adding, because 
the Java SDK leverages *mutiny-zero* as its reactive foundation, a
decision that reflects our commitment to framework-agnostic excellence.

> [Mutiny Zero](https://smallrye.io/smallrye-mutiny-zero/latest/) is a minimal API for creating reactive streams-compliant

Reviewer 2: https://smallrye.io/smallrye-mutiny-zero/latest/[Mutiny Zero]

=== *Community-Driven Innovation*

The Java SDK represents a perfect example of open-source collaboration
in action. Originally developed by our talented team, this
implementation demonstrates the power of community-driven innovation.

What started as an external contribution has now become an official part
of the A2A project repository, showcasing how the ecosystem can rapidly
evolve through diverse contributions. This is precisely the kind of
collaborative development that will accelerate A2A adoption and
innovation.

Ready to dive in? Here’s your roadmap:

[arabic]
. *Explore the SDK*: Visit
https://github.com/a2aproject/a2a-java[github.com/a2aproject/a2a-java]
to examine the implementation +
. *Study Real Examples*: Check out the
https://github.com/a2aproject/a2a-samples/tree/main/samples/multi_language/python_and_java_multiagent[multi-language
samples] to see interoperability in action +
. *Join the Community*: Connect with fellow developers in the A2A
ecosystem +
. *Start Building*: Begin prototyping your first multi-language agent
team

=== *The Bigger Picture: Collaborative Intelligence*

The A2A protocol aims to break down the silos that currently limit the

> potential of artificial intelligence by providing a common language for

Reviewer: ```suggestion
potential of AI infused applications by providing a common language for
```

I don't like giving too much credit to AI itself/alone :)

---
layout: post
title: 'Quarkus and WildFly teams from Red Hat collaborating with Google on launch of Agent2Agent Java SDK'
date: 2025-06-27
tags: release
synopsis: 'We’re thrilled to announce the launch of the Agent2Agent (A2A) Java SDK, made possible by WildFly and Quarkus teams’ contribution.'
author: maeste
---
:imagesdir: /assets/images/posts/hibernate7
ifdef::env-github,env-browser,env-vscode[:imagesdir: ../assets/images/posts/hibernate7]

The agent revolution just took a massive leap forward! Following the

> recent landmark https://developers.googleblog.com/en/google-cloud-donates-a2a-to-linux-foundation/[announcement that Google has donated the Agent2Agent
> (A2A) protocol to the Linux Foundation], we’re thrilled to announce the

Reviewer: ```suggestion
recent landmark https://developers.googleblog.com/en/google-cloud-donates-a2a-to-linux-foundation/[announcement] that Google has donated the Agent2Agent
(A2A) protocol to the Linux Foundation, we’re thrilled to announce the
```

to not have a link that crosses multiple lines

> launch of the *https://github.com/a2aproject/a2a-java[A2A Java SDK]*, built by WildFly and Quarkus teams in close collaboration, and now contributed to the official A2A project.

Reviewer: ```suggestion
launch of the https://github.com/a2aproject/a2a-java[A2A Java SDK], built by WildFly and Quarkus teams in close collaboration, and now contributed to the official A2A project.
```

**'s makes the link broken so removed it
